utils.go

```diff
@@ -182,14 +182,14 @@ func DefaultStoreOptions(rootless bool, rootlessUid int) (StoreOptions, error) {
 		err                      error
 	)
 	storageOpts := defaultStoreOptions
-	if rootless {
+	if rootless && rootlessUid != 0 {
 		storageOpts, err = getRootlessStorageOpts(rootlessUid)
 		if err != nil {
 			return storageOpts, err
 		}
 	}
 
-	storageConf, err := DefaultConfigFile(rootless)
+	storageConf, err := DefaultConfigFile(rootless && rootlessUid != 0)
```

Comment:
Do we need this fix for all callers of `DefaultConfigFile()`? If so, I think we need to fix it there.

Reply:
The only place other then Storage where this is currently called is in libpod.

./libpod/info.go:	configFile, err := storage.DefaultConfigFile(rootless.IsRootless())
./libpod/runtime.go:		storageConfFile, err := storage.DefaultConfigFile(rootless.IsRootless())

Reply:
good point, but we will need to change `DefaultConfigFile()` to accept another argument as the `rootlessUid` is not known

Reply:
Can't we just fix the callers to call this correctly.  IE Only call it when you are truly in rootless mode.
I am thinking the problem here is in the callers telling us they are in rootless mode,when they are running as root.

Reply:
Anyways I think this PR is correct and solves the problem we are seeing.  We need to fix libpod to figure out whether it is REALY in rootless mode, or just running within a User Namespace, Which are really two different things.